pkg/model/openstackmodel/servergroup.go

```diff
@@ -141,8 +143,11 @@ func (b *ServerGroupModelBuilder) buildInstances(c *fi.ModelBuilderContext, sg *
 		}
 		c.AddTask(portTask)
 
+		igMeta[openstack.TagKopsName] = fi.StringValue(instanceName)
```

Comment:
The issue is probably this one. igMeta is defined outside of the loop, so we are setting the same metadata on all instances.

Reply:
yeah it seems that if we have more than 1 instance in same instancegroup, the instancename will be same in all of them

Reply:
but the weird thing is that it works when I execute this against openstack.. blah

Reply:
% kubectl get nodes
NAME                   STATUS   ROLES    AGE     VERSION
master-zone-1-prb0qx   Ready    master   7m43s   v1.19.6
nodes-zone-1-epkztj    Ready    node     6m29s   v1.19.6
nodes-zone-1-ldgep8    Ready    node     6m26s   v1.19.6

nodes-zone-1-epkztj 
metadata
KopsName nodes-zone-1-2-calc-k8s-local

nodes-zone-1-ldgep8 
metadata
KopsName nodes-zone-1-1-calc-k8s-local

Reply:
Not sure. Probably some serialisation/deserialisation that does not happen outside of tests.